sky/execution.py

```diff
@@ -554,16 +554,15 @@ def spot_launch(
     assert dag.is_chain(), ('Only single-task or chain DAG is '
                             'allowed for spot_launch.', dag)
 
-    dag_utils.infer_and_fill_dag_name(dag)
+    dag_utils.maybe_infer_and_fill_dag_and_task_names(dag)
 
     task_names = set()
     for task_ in dag.tasks:
-        if task_.name is not None:
-            if task_.name in task_names:
-                raise ValueError(
-                    f'Task name {task_.name} is duplicated in the dag.')
-            task_names.add(task_.name)
-    for task_id, task_ in enumerate(dag.tasks):
+        if task_.name in task_names:
+            raise ValueError(
+                f'Task name {task_.name} is duplicated in the dag.')
```

Comment:
```suggestion
            raise ValueError(
                f'Task name {task_.name!r} is duplicated in the DAG. Either change task names to be unique, or specify the DAG name only and comment out the task names (so that they will be auto-generated) .')
```